src/utils.ts

```diff
@@ -130,11 +130,19 @@ export function mockGlobalTSConfigSchema(globals: any) {
     { __TS_CONFIG__: config };
 }
 
+const tsConfigCache: { [key: string]: any } = {};
 export function getTSConfig(globals, collectCoverage: boolean = false) {
   let config = getTSConfigOptionFromConfig(globals);
   const skipBabel = getTSJestConfig(globals).skipBabel;
   const isReferencedExternalFile = typeof config === 'string';
 
+  // check cache before resolving configuration
+  // NB: config is a string unless taken from __TS_CONFIG__, which should be immutable (and is deprecated anyways)
+  const tsConfigCacheKey = JSON.stringify([skipBabel, collectCoverage, isReferencedExternalFile ? config : undefined]);
```

Comment:
`JSON.stringify` is available without imports and will produce a consistent, unique key. It didn't seem worth the extra effort to generate a uniform key via `crypto`. Can do that if it's desired, though.